internal/controller/deployment_to_pdb_controller.go

```diff
@@ -65,6 +68,36 @@ func (r *DeploymentToPDBReconciler) handleDeploymentReconcile(ctx context.Contex
 
 			// PDB already exists, nothing to do
 			log.Info("PodDisruptionBudget already exists", "namespace", pdb.Namespace, "name", pdb.Name)
+			pdbWatcher := &myappsv1.PDBWatcher{}
+			e := r.Get(ctx, types.NamespacedName{Name: pdb.Name, Namespace: pdb.Namespace}, pdbWatcher)
+			if e == nil {
+				// if pdb exists get pdbWatcher --> compare targetGeneration field for deployment if both not same deployment was not changed by pdb watcher
+				// update pdb minReplicas to current deployment replicas
+				log.Info("deployment replicas got updated", " pdbWatcher.Status.TargetGeneration", pdbWatcher.Status.TargetGeneration, "deployment.Generation", deployment.GetGeneration())
+				if pdbWatcher.Status.TargetGeneration != deployment.GetGeneration() {
+					_, scaleDownAnnotationExists := deployment.Annotations["NewReplicasAfterScaledDownByPdbWatcher"]
```

Comment:
I _think_ you only the scaled up one. 

Want to use a consts. Suggest maybe calling it. evictionSurgeReplicas (later we should suffix it with like evictionautoscaler.azure.io)